plugin/trino-influxdb/src/main/java/io/trino/plugin/influxdb/InfluxConfig.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.trino.plugin.influxdb;
+
+import io.airlift.configuration.Config;
+import io.airlift.configuration.ConfigSecuritySensitive;
+import io.airlift.units.Duration;
+import io.airlift.units.MinDuration;
+import jakarta.validation.constraints.AssertFalse;
+import jakarta.validation.constraints.NotNull;
+import jakarta.validation.constraints.Pattern;
+
+import java.util.Optional;
+
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+public class InfluxConfig
+{
+    private String endpoint;
+    private String username;
+    private String password;
+    private Duration connectTimeOut = new Duration(10, SECONDS);
+    private Duration readTimeOut = new Duration(60, SECONDS);
+
+    @NotNull
+    @Pattern(message = "Invalid endpoint. Expected http:// ", regexp = "^http://.*")
```

Comment:
I don't understand why https is prohibited. Also, it would be nice to allow user-password only when https is enabeld (=disallow it in case of http).